### ISSUE TITLE
Needs PubSub Admin otherwise fails with 403

### ISSUE THREAD

**Issue by livioso** [logstash-plugins/logstash-input-google_pubsub#9]
Hi,

I'm curious why the service account needs to have Pub/Sub Admin rights in order for this Plugin to run. If I create a service account with less rights than Admin I get `Error: Error 403: User not authorized to perform this action.` during startup. If this is intended behaviour I'll update the readme to include this in the getting started section. :)

Thanks,
Livio

**Comment by ScottBrenner** [comment]
This is an issue with GCP's IAM roles, not this plugin per se.

**Comment by Zhomart** [comment]
Gem `google-cloud-pubsub` uses better API and works only with "Pub/Sub Views" and "Pub/Sub Publisher" rights.

**Comment by wleese** [comment]
It'd be nice if this gets fixed some time. But in the mean time, instead of giving the service account the Pub/Sub Admin role, you can create a custom role with the following permissions:

```
    pubsub.subscriptions.consume
    pubsub.subscriptions.create
    pubsub.subscriptions.get
    pubsub.subscriptions.getIamPolicy
    pubsub.subscriptions.list
    resourcemanager.projects.get
```

**Comment by hakenmt** [comment]
Also encountering this issue. The plugin shouldn't attempt to create the subscription and fail if the subscription already exists. It should be adequate to just provide the Pub/Sub Subscriber role on the specified subscription that already exists and Pub/Sub Viewer on the project.

**Comment by jakelandis** [comment]
This is addressed as of [1.2.0](https://github.com/logstash-plugins/logstash-input-google_pubsub/blob/master/CHANGELOG.md#120)

Thanks everyone for the feedback and contributions !
